pkg/plugin/response.go

```diff
@@ -277,3 +298,132 @@ func labelsToJSON(labels data.Labels) (json.RawMessage, error) {
 
 	return b, nil
 }
+
+const (
+	vector, matrix, scalar = "vector", "matrix", "scalar"
+)
+
+// Result represents timeseries from query
+type Result struct {
+	Labels Labels  `json:"metric"`
+	Values []Value `json:"values"`
+	Value  Value   `json:"value"`
+}
+
+// Value represents timestamp and value of the timeseries
+type Value [2]interface{}
+
+// Labels represents timeseries labels
+type Labels map[string]string
+
+// Data contains fields
+// ResultType which defines type of the query response
+// Result resents json of the query response
+type Data struct {
+	ResultType string          `json:"resultType"`
+	Result     json.RawMessage `json:"result"`
+}
+
+// Response contains fields from query response
+type Response struct {
+	Status string `json:"status"`
+	Data   Data   `json:"data"`
+}
+
+type promInstant struct {
```

Comment:
`promInstant` has identical structure as `promRange`. Are you sure these need to be two different structs? Should we change the name of the struct to remove `prom` connotation as it seems irrelevant here and rather is a result of copypaste

Reply:
used one struct and simplify a little `getDataFrames` method